spanner/transaction.go

```diff
@@ -135,26 +142,31 @@ func (t *txReadOnly) ReadWithOptions(ctx context.Context, table string, keys Key
 	}
 	index := ""
 	limit := 0
+	var ro *sppb.RequestOptions
 	if opts != nil {
 		index = opts.Index
 		if opts.Limit > 0 {
 			limit = opts.Limit
 		}
+		if opts.Priority != sppb.RequestOptions_PRIORITY_UNSPECIFIED {
```

Comment:
nits: can this be replaced by `createRequestOptions()`? The logic looks a bit similar.

Reply:
Good point, and that also applies to a number of other places as well. I've introduced a generic `requestPrioritizer` interface that can be used to generate the request options. This might some tweaking when the tagging feature can be merged as well, as both will generate `RequestOptions`, but as it is all internal interfaces it should not really be a problem.